[#57402] Fix Active Record Pool Reaper thread leak after `Parallelization#shutdown` (#57406)
rails/rails: activerecord/CHANGELOG.md

> *   Fix AR Pool Reaper thread leak after `Parallelization#shutdown`.

Reviewer: ```suggestion
*   Fix Active Record Pool Reaper thread leak after `Parallelization#shutdown`.
```

>     After parallelized test runs, the parent process leaked the AR Pool Reaper

Reviewer: ```suggestion
    After parallelized test runs, the parent process leaked the Active Record Pool Reaper
```

    thread, holding open connection pools and file descriptors for up to
    `reaping_frequency` seconds (or indefinitely if never reaped).

    Three fixes: `Parallelization#shutdown` now calls `run_cleanup_hooks`;

>     AR registers a cleanup hook that discards all connection pools; and

Reviewer: ```suggestion
    Active Record registers a cleanup hook that discards all connection pools; and
```
